### PR TITLE
Add htaccess to redirect if not found to index

### PR DESCRIPTION
**Pull request by the-hotmann**
If (file not found or directory not found) 
then > redirect to index.html

Related to https://github.com/shlinkio/shlink-web-client/issues/196

### CODE REVIEW

**Comment by acelaya** [comment]
Awesome! Thanks a lot @MartinH0 😁

Just have two comments.

* The htaccess file should be located inside the `public` directory, otherwise the build process won't include it in the final dist file.
* This rule will have a side effect. When accidentally introducing a typo on an asset route (like an image), Apache will serve the index file instead, making it hard to detect. Ideally, when the path has an extension, Apache should continue returning a 404, and only for paths without extension (which are usually handled client-side) it should fall back to the index file.

For the second point, you can take a look at how it's done for the nginx inside the docker image https://github.com/shlinkio/shlink-web-client/blob/master/config/docker/nginx.conf

**Comment by the-hotmann** [comment]
Wait. I will optimize it. So just files without file extensions will be redirected

**Comment by the-hotmann** [comment]
Now you can merge if you like it.

**Comment by acelaya** [comment]
Thanks for the contribution! I will test it later and merge it if everything goes as expected.

File: config/docker/nginx.conf [hunk 1]
```diff
@@ -5,7 +5,7 @@ server {
     index index.html;
 
     # When requesting static paths with extension, try them, and return a 404 if not found
-    location ~ .+\.(css|js|html|png|jpg|jpeg|gif|bmp|ico|json|csv|otf|eot|svg|svgz|ttf|woff|woff2|ijmap|pdf|tif|map) {
+    location ~* .+\.(css|js|html|png|jpe?g|gif|bmp|ico|json|csv|otf|eot|svg|svgz|ttf|woff|woff2|ijmap|pdf|tif|map) {
```
Comment (acelaya) [inline config/docker/nginx.conf:8]: What does this extra `*` do? What's the difference between `location ~*` and `location ~`?

Comment (acelaya) [reply config/docker/nginx.conf:8]: Never mind. Just found the explanation here https://www.digitalocean.com/community/tutorials/understanding-nginx-server-and-location-block-selection-algorithms :sweat_smile: 

Good catch.

**Review by acelaya — APPROVED** [review]
Just tested all changes, and they work as expected. Good job :)

**Pull request merged**